### PR TITLE
[React-native]: App tutorial

### PR DESCRIPTION
**Pull request by brunokiafuka**
This PR adds the app tutorials on the React Native applications, also contains a fix on the app shortcuts (namely `archive` and `delete`).

- Also shimmed `sessionStorage`.

### CODE REVIEW

File: src/components/Tutorial/Tutorial2StepContext2.native.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+const { smallText, italic } = commonStyles
+
+// eslint-disable-next-line jsdoc/require-jsdoc
+const Tutorial2StepContext2 = ({ tutorialChoice, rootChildren, cursor }: IComponentProps) => (
+  <Fragment>
+    <Text style={smallText}>
+      Now add a thought with the text "{TUTORIAL_CONTEXT[tutorialChoice]}"{' '}
+      <Text style={[smallText, italic]}>within</Text> “{TUTORIAL_CONTEXT2_PARENT[tutorialChoice]}”.
+    </Text>
+    {
```
Comment (Foosballfan) [inline src/components/Tutorial/Tutorial2StepContext2.native.tsx:28]: all these ternarys are real ugly. would be nice to split the logic outside of the jsx

File: src/components/Tutorial/Tutorial2StepContext2Subthought.native.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+const { smallText, italic } = commonStyles
+
+// eslint-disable-next-line jsdoc/require-jsdoc
+const context2SubthoughtCreated = ({ rootChildren, tutorialChoice }: IComponentProps) => {
+  const state = store.getState()
+  // e.g. Work
+  return (
+    rootChildren.find(child => child.value.toLowerCase() === TUTORIAL_CONTEXT2_PARENT[tutorialChoice].toLowerCase()) &&
```
Comment (Foosballfan) [inline src/components/Tutorial/Tutorial2StepContext2Subthought.native.tsx:39]: theres opportunity for a single util function which compares two values toLowerCase

File: src/components/Tutorial/TutorialNavigationNext.native.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+  return [
+    TUTORIAL_STEP_START,
+    TUTORIAL_STEP_SUCCESS,
+    TUTORIAL2_STEP_START,
+    TUTORIAL2_STEP_CONTEXT_VIEW_OPEN,
+    TUTORIAL2_STEP_CONTEXT_VIEW_EXAMPLES,
+    TUTORIAL2_STEP_SUCCESS,
+  ].includes(tutorialStep) ||
+    (tutorialStep === TUTORIAL_STEP_AUTOEXPAND && Object.keys(expanded).length === 0) ||
```
Comment (Foosballfan) [inline src/components/Tutorial/TutorialNavigationNext.native.tsx:56]: this is entirely unreadable

File: src/components/Tutorial/TutorialStepSubThoughtEnter.native.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+const { smallText, italic } = commonStyles
+// eslint-disable-next-line jsdoc/require-jsdoc
+const TutorialStepSubThoughtEnter = ({ cursor }: IComponentProps) => (
+  <Fragment>
+    <Text style={smallText}>
+      As you can see{isTouch ? ' (if you scroll down)' : ''}, the new thought
+      {cursor && cursor.length > 1 && headValue(cursor).length > 0 ? (
+        <Fragment> "{ellipsize(headValue(cursor))}"</Fragment>
```
Comment (Foosballfan) [inline src/components/Tutorial/TutorialStepSubThoughtEnter.native.tsx:19]: dont think this fragment is necessary

**Review by Foosballfan — APPROVED** [review]

**Review by shresthabijay — APPROVED** [review]
Looks good to me!

**Pull request merged**